src/algorithms.jl

```diff
@@ -0,0 +1,78 @@
+abstract type AbstractEnclosureAlgorithm end
+
+"""
+   NaturalEnclosure <: AbstractEnclosureAlgorithm
+
+Data structure to compute the range of `f` over `X` using natural enclosure, that is simply
+evaluates `f(X)` using interval arithmetic.
+"""
+struct NaturalEnclosure <: AbstractEnclosureAlgorithm end
+
+"""
+    MooreSkelboeEnclosure{T} <: AbstractEnclosureAlgorithm
+
+Data structure to compute the range of `f` over `X` using the Moore-Skelboe algorithm, which
+computes rigorously the global minimum and maximum of the function.
+See `IntervalOptimisation.jl` for more details.
+
+### Fields
+
+- `structure` -- (default: `HeapedVector`) the way in which vector elements
+                 are kept arranged; possible options are `HeapedVector` and `SortedVector`
+- `tol`       -- (default `1e-3`) tolerance to which the optima are computed
+
+### Notes
+
+To use this algorithm, you need to import `IntervalOptimisation.jl`
+"""
+struct MooreSkelboeEnclosure{T} <: AbstractEnclosureAlgorithm
+    structure::T
+    tol::Float64
+end
+
+"""
+    TaylorModelsEnclosure <: AbstractEnclosureAlgorithm
+
+Data structure to compute the range of `f` over `X` using taylor models.
+See `TaylorModels.jl` for more details.
+
+### Fields
+
+- `order`     -- (default: `10`) order of the taylor model used to compute
+                 an enclosure of `f` over `dom`
+- `normalize` -- (default: `true`) if `true`, normalize the taylor model
+                 on the unit symmetric box around the origin
+
+"""
+struct TaylorModelsEnclosure <: AbstractEnclosureAlgorithm
+    order::Int
+    normalize::Bool
+end
+
+"""
+    SumOfSquaresEnclosure{T} <: AbstractEnclosureAlgorithm
+
+Data structure to compute the range of `f` over `X` using the sum of squares algorithm.
+See `SumOfSquares.jl` for more details
+
+### Fields
+- `backend` -- backend used to solve the optimization problem, a list of available backends
+             can be found at
+- `order` -- (default `5`), maximum degree of the SDP relaxation
+- `quiet` -- (default `false`), I don't know what this is
```

Comment:
@mforets?

Reply:
I think that was there to choose verbosity, see [set_silent](https://jump.dev/JuMP.jl/stable/reference/models/#JuMP.set_silent)

Reply:
a priori i would choose `quiet = true` by default

Reply:
I'm a little confused, if you look at the implementation of  `enclose_SumOfSquares`, it propagates the `kwargs` until `new_sos`, but then only `QUIET` is taken into account (see [here](https://github.com/JuliaReach/RangeEnclosures.jl/blob/master/src/sdp.jl#L20-L29) ), I'm not familiar with the package, does `SOSModel` have other kwargs? is `quiet` really the only one that we want to allow the user to set? I see two options

1. `QUIET` is the only we care, then have `quiet` in the field of the struct and don't pass other kwargs
2. `SOSModel` has several kwargs and we want the user to configure those. At this point I don't think `quiet` is special, it can be removed from the fields and we can have something like
```julia
function enclose(f, X, sos::SumOfSquares; backend, kwargs....)
  model = SOSModel(backend; kwargs...)
```
and document you can pass extra keyword arguments that will be passed to `SOSModel`

I think maybe the second one makes slightly more sense?

Reply:
i think that for true algorithm-agnostic  verbosity control, `set_silent` / `unset_silent` should be used

here i suggest to remove the field in the struct, better to keep them minimal, i would say that controlling verbosity an option that applies to all solvers hence we may want to pass that as an extra kwarg in `enclose`, so i would leave that for now